src/lib/converter/iPUZtoJSON.js

```diff
@@ -0,0 +1,61 @@
+function convertCluesArray(initialCluesArray) {
+  const finalCluesArray = [];
+
+  for (let i = 0; i < initialCluesArray.length; i++) {
+    let item = initialCluesArray[i];
+    let number, stringClue;
+    if (Array.isArray(item)) {
+      number = parseInt(item[0]);
+      stringClue = item[1];
+    } else {
+      number = parseInt(item.number);
+      stringClue = item.clue;
+    }
+    finalCluesArray[parseInt(number)] = stringClue;
+  }
+
+  return finalCluesArray;
+}
+
+export default function iPUZtoJSON(readerResult) {
+  const jsonFromReader = JSON.parse(new TextDecoder().decode(readerResult));
+  const grid = jsonFromReader.solution.map((row) =>
+    row.map((cell) => (cell === null || cell === '#' ? '.' : cell))
+  );
+  const info = {
+    type: grid.length > 10 ? 'Daily Puzzle' : 'Mini Puzzle',
+    title: jsonFromReader.title || '',
+    author: jsonFromReader.author || '',
+    description: jsonFromReader.description || '',
+  };
+  const circles = [];
+  const shades = [];
```

Comment:
As a side note I don't think shades actually ever worked for normal `puz` files currently -- I believe it will always read as circles.

For `ipuz`, I think we can read `shades` in if there is a `cell.style.color`, but I couldn't find any good examples of this when I used tools like Crossword Scraper to download `ipuz` files, it always just set the `cell.style.shapebg`

Reply:
i see, that's interesting. 
re: `shades` i don't remember exactly, but i think there were some non-standard puz files that did set this field. in any case those are probably irrelevant so i agree with your comment.

re: ipuz shades, sounds good to me for now. feel free to make a follow-up PR if you want to update this logic for ipuz!